CONTRIBUTING.md

```diff
@@ -19,422 +19,442 @@ under the License.
+[ESLint](https://eslint.org/) as a tool for identifying and reporting on patterns in JavaScript, which can be 
+used by running any of the following commands:
 
+```bash
+# Check JS code in .js and .html files, and report any errors/warnings
+npm run lint
+
+# Check JS code in .js and .html files, report any errors/warnings and fix them if possible
+npm run lint:fix
 ```
-./run_unit_tests.sh tests.core:CoreTest -s --logging-level=DEBUG
-```
 
-or a single test method:
+# Developing Airflow
+
+## Running unit tests
+
+To run tests locally, once your unit test environment is set up you should be able to run 
+``./run_unit_tests.sh`` at will. For example, in order to just execute the "core" unit tests, run the 
+following:
 
+```bash
+./run_unit_tests.sh tests.core:CoreTest -s --logging-level=DEBUG
 ```
+
+Or a single test:
+```bash
 ./run_unit_tests.sh tests.core:CoreTest.test_check_operators -s --logging-level=DEBUG
 ```
-or another example:
-```
+
+Another example:
+```bash
 ./run_unit_tests.sh tests.contrib.operators.test_dataproc_operator:DataprocClusterCreateOperatorTest.test_create_cluster_deletes_error_cluster  -s --logging-level=DEBUG
 ```
 
 To run the whole test suite with Docker Compose, do:
-
-```
+```bash
 # Install Docker Compose first, then this will run the tests
 docker-compose -f scripts/ci/docker-compose.yml run airflow-testing /app/scripts/ci/run-ci.sh
 ```
 
-Alternatively, you can also set up [Travis CI](https://travis-ci.org/) on your repo to automate this.
-It is free for open source projects.
+## Setting up your own Travis CI
 
-Another great way of automating linting and testing is to use [Git Hooks](https://git-scm.com/book/uz/v2/Customizing-Git-Git-Hooks). For example you could create a `pre-commit` file based on the Travis CI Pipeline so that before each commit a local pipeline will be triggered and if this pipeline fails (returns an exit code other than `0`) the commit does not come through.
-This "in theory" has the advantage that you can not commit any code that fails that again reduces the errors in the Travis CI Pipelines.
+Every PR made to Airflow is automatically passed through the Airflow CI pipeline, which runs on Travis. If you
+want to run the same pipeline without having to submit a PR, you can set up your own Travis CI.
 
-Since there are a lot of tests the script would last very long so you probably only should test your new feature locally.
+Register your Airflow fork in Travis ([guide](https://docs.travis-ci.com/user/tutorial)) and the pipeline will
+run on every push. 
 
-The following example of a `pre-commit` file allows you..
-- to lint your code via flake8
-- to test your code via nosetests in a docker container based on python 2
-- to test your code via nosetests in a docker container based on python 3
+Note that travis-ci.org is legacy; make sure you register your fork on
+[travis-ci.com](https://travis-ci.com).
 
-```
```

Comment:
Instead of removing this. I think we can update the pre-commit script to run pylint, mypy and flake8 and move it under `Tool specific tips & tricks` section. Or we can just create a separate `pre-commit` file.

Reply:
@milton0825 - in the (ready-to-merge finally) #4938 we have scripts to run pylint/mypy etc. automatically inside a downloaded Docker image. I think as next step we can add pre-commit hooks integration with https://pre-commit.com/  - this way we can add an easy way to configure pre-commit hooks and run them.

Reply:
That is awesome. Looking forward to it.